src/Microsoft.DotNet.Build.Tasks/Targets/packageresolve.targets

```diff
@@ -103,4 +103,20 @@
     <Message Text="Excluding @(_ReferenceFileNamesToRemove);@(_NoneFileNamesToRemove) from package references since the same file is provided by a project refrence."
              Condition="'@(_ReferenceFileNamesToRemove)' != '' or '@(_NoneFileNamesToRemove)' != ''"/>
   </Target>
+
+  <!-- the helix test infrastructure uses the generated assembly list to determine the runtime binaries to restore -->
+  <Target Name="GenerateAssemblyListTxt" AfterTargets="ResolveNuGetPackages" Condition="'$(PrereleaseResolveNuGetPackages)'=='true' and '$(IsTestProject)'=='true'">
+    <Message Text="Generating $(OutDir)AssemblyList.txt" />
+
+    <ItemGroup Condition="Exists('$(ProjectLockJson)')">
+      <AssemblyPaths Include="$([MSBuild]::MakeRelative($(PackagesDir), %(CopyLocalItems.FullPath)))" />
+    </ItemGroup>
+
+    <WriteLinesToFile Condition="Exists('$(ProjectLockJson)')"
```

Comment:
Can you help me understand why it needs a separate list as opposed to resolving these and copying them to a directory and zipping that up?

Also is this even the complete list? I would expect it would need more things like the test-runtime file set as well.